basescript/basescript.py

```diff
@@ -72,21 +72,16 @@ def start(self):
         # invoke the appropriate sub-command as requested from command-line
         try:
             self.args.func()
-        except SystemExit as e:
-            if e.code != 0:
```

Comment:
@RamanjaneyuluIdavalapati Why was trapping `SystemExit` removed?

Reply:
@prashanthellina This is because I don't see any benefit of using SystemExit. Because you are raising again if it is non zero.

Without SystemExit also the same happens, no?

Reply:
@RamanjaneyuluIdavalapati If in `run` function we do "sys.exit(0)", can you paste the output logs of the script before your change and after?

Reply:
Code
```python
import sys 

from basescript import BaseScript

class Test(BaseScript):
    def run(self):
        sys.exit(0)

if __name__ == '__main__':
    Test().start()
```

Before the change: python sys_test.py --log-level DEBUG run
```
2020-03-17T11:42:21.545249Z [debug    ] basescript init                commands=run debug=False func=run host=hp23.servers.nferx.com id=20200317T114221_5cfe4314684411ea9e86309c23ac5803 log_file=None log_format=None log_level=DEBUG metric_grouping_interval=1 minimal=False name=sys_test.py quiet=False type=log
2020-03-17T11:42:21.545503Z [debug    ] exited_successfully            host=hp23.servers.nferx.com id=20200317T114221_5cfe4315684411ea9e86309c23ac5803 name=sys_test.py type=log
```

After the change: python sys_test.py --log-level DEBUG run
```
2020-03-17T11:39:17.647406Z [debug    ] basescript init                commands=run debug=False func=run host=hp23.servers.nferx.com id=20200317T113917_ef61ad6e684311ea9e86309c23ac5803 log_file=None log_format=None log_level=DEBUG metric_grouping_interval=1 minimal=False name=sys_test.py quiet=False type=log
```

Reply:
Hmm, I can see the difference. One more log "exited_successfully".

If I add back the "SystemExit" then I have to move down the log which I have changed in this commit.